### DISCUSSION TITLE
Support for Gremlin

### DISCUSSION THREAD

**Discussion by hypatia1729** [ChilliCream/hotchocolate#4086] [Ideas]
Hi,

I see that you've got work in progress code to support Neo4J; I was wondering if there's anything on the roadmap for Gremlin. It seems that the efficiency of the Gremlin engine for graph type queries is a natural fit for GQL projections, so that a single Gremlin query would be generated to take advantage of the graph engine's optimisations. It'd presumably involve edge classes as 'joins' rather than foreign keys, so the class model would be different, maybe even allowing first class support for vertex types and edge types, but totally workable I imagine.

We're currently doing extensive work on a Cosmos DB Gremlin database, and we're probably going to end up extending HotChocolate to get something out for GQL in a short time frame, but it'd almost certainly be substandard and nowhere near as complete as anything you guys would develop, so would love to know if you're got anything on the horizon.

Thanks, love your work,
Sonja

**Comment by jbockle** [comment]
Having gremlin support would be really great - I'd like to make an integration with it but I don't really know where to start, is there any general guidance anyone can provide?  Having some general documentation on how integrations work/considerations/etc would be really helpful.
